crates/rspack_plugin_javascript/src/dependency/esm/harmony_export_imported_specifier_dependency.rs

```diff
@@ -1350,39 +1345,36 @@ pub struct StarReexportsInfo {
 }
 
 /// return (names, dependency_indices)
-fn determine_export_assignments(
-  module_graph: &ModuleGraph,
-  mut dependencies: Vec<DependencyId>,
+fn determine_export_assignments<'a>(
+  module_graph: &'a ModuleGraph,
+  dependencies: &[DependencyId],
   additional_dependency: Option<DependencyId>,
-) -> (Vec<Atom>, Vec<usize>) {
-  if let Some(additional_dependency) = additional_dependency {
-    dependencies.push(additional_dependency);
-  }
-
+) -> (Vec<&'a Atom>, Vec<usize>) {
   // https://github.com/webpack/webpack/blob/ac7e531436b0d47cd88451f497cdfd0dad41535d/lib/dependencies/HarmonyExportImportedSpecifierDependency.js#L109
-  // js `Set` keep the insertion order, use `IndexSet` to align there behavior
-  let mut names: IndexSet<Atom, BuildHasherDefault<FxHasher>> = IndexSet::default();
-  let mut dependency_indices = vec![];
+  // js `Set` keep the insertion order
+  let mut names = Vec::new();
+  let mut hints = HashSet::default();
+  let mut dependency_indices =
+      Vec::with_capacity(dependencies.len() + additional_dependency.is_some() as usize);
 
-  for dependency in dependencies.iter() {
-    dependency_indices.push(names.len());
+  for dependency in dependencies.iter().chain(additional_dependency.iter()) {
     if let Some(module_identifier) = module_graph.module_identifier_by_dependency_id(dependency) {
       let exports_info = module_graph.get_exports_info(module_identifier);
       for export_info_id in exports_info.exports.values() {
         let export_info = module_graph.get_export_info_by_id(export_info_id);
         // SAFETY: This is safe because a real export can't export empty string
-        let export_info_name = export_info.name.clone().unwrap_or_default();
+        let export_info_name = export_info.name.as_ref().expect("export name is empty");
```

Comment:
I'm not sure we need to be consistent with original behavior.

Reply:
We can't use Cow, but I can use static empty Atom instead. but it would be better if it could be guaranteed not to be None.

Reply:
Expect other_export_info, I don't see any where that export_info.name could be None, so it looks ok to me